jaypeasee/whats-cookin
# Feature - User Generation

# Pull Request Template:

#### What's this PR do?
1. Randomly creates a new `User` on load in the DOM.
1. Instantiates the `Pantry` Class within the `User`
1. Adjusts the recipe display to have a more uniform grid.
1. Creates testing for filtering recipes by tags
1. Passes tests for filtering recipes by tags
1. Creates tag buttons that affect the DOM and reflect filtered tags.
1. Creates a test for filtering recipe list by ingredient ID
1. Passes tests for filtering recipes
1. Adds functionality to search bar for searching by ingredients.
1. Converts user input from a string to corresponding ingredient ID to pass into filtering method.
1. Fixes previous tests that started failing after the search one passed on sad path testing.


#### Where should the reviewer start?
* Review `user-test.js`
* Review `user-js`
* Review `scripts.js`


#### How should this be manually tested?
1. Click on the filter tags to see how the recipe view changes.
1. Enter some common ingredient names in the searchbar to see how the recipe view changes

#### Any background context you want to provide?
We had specific help from JP's mentor, working through the `searchByIngredients()` method

#### Screenshots (if appropriate):
<img width="1433" alt="Screen Shot 2020-10-14 at 7 14 41 PM" src="https://user-images.githubusercontent.com/59029768/96061871-861b2d80-0e51-11eb-8e12-dc9280fff36f.png">
